python/plugins/processing/gui/OutputSelectionPanel.py

```diff
@@ -100,6 +106,20 @@ def selectOutput(self):
 
             popupMenu.exec_(QCursor.pos())
 
+    def showExpressionsBuilder(self):
+        context = QgsExpressionContext()
+        context.appendScope(QgsExpressionContextUtils.globalScope())
+        scope = QgsExpressionContextUtils.projectScope()
+        self.addVariablesToScope(scope)
```

Comment:
same here - this should add the projectScope intact, then create a new blank scope for the processing variables